### ISSUE TITLE
[ERROR] precache error

### ISSUE THREAD

**Issue by reid123455** [shanapu/MyStore#11]
Sorry for my bad English first
I tried various Store plugins, but some of those have not maintained anymore
I got the same problem, if I changed model, plugin would cause server crashing
After crashing I got an Error window like
```
1/- player: UTIL_SetModel: not parecached:
model/players/custom_player/whatever/whatever.mdl
```
this is my log file with last two errors
most are about CreateTimer
```
L 12/13/2019 - 04:15:12: [SM] Call stack trace:
L 12/13/2019 - 04:15:12: [SM]   [0] CreateTimer
L 12/13/2019 - 04:15:12: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn
L 12/13/2019 - 04:15:35: [SM] Exception reported: Could not create timer, no more handles
L 12/13/2019 - 04:15:35: [SM] Blaming: mystore_item_playermodel.smx
L 12/13/2019 - 04:15:35: [SM] Call stack trace:
L 12/13/2019 - 04:15:35: [SM]   [0] CreateTimer
L 12/13/2019 - 04:15:35: [SM]   [1] Line 253, addons/sourcemod/scripting/mystore_item_playermodel.sp::SetClientModel
L 12/13/2019 - 04:15:35: [SM]   [2] Line 181, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_Equip
L 12/13/2019 - 04:15:35: [SM]   [4] Call_Finish
L 12/13/2019 - 04:15:35: [SM]   [5] Line 3509, addons/sourcemod/scripting/mystore_core.sp::UseItem
L 12/13/2019 - 04:15:35: [SM]   [6] Line 1728, addons/sourcemod/scripting/mystore_core.sp::MenuHandler_Preview
```
and this is my Accelerator crash log
https://crash.limetech.org/dteks7vsvnla

# Environment
  * Operating System version:  Windows 10 Pro
  * Current SourceMod version: 1.10 (last version)
  * Current Metamod:Source snapshot version: 1.11
  * Current MyStore version: MyStore-0.1.35-5c35ac3

Please help
If I lose some information, please tell me

**Comment by dzinks2009** [comment]
Did you find a fix?

**Comment by Ayrton09** [comment]
> Did you find a fix?

nothing

**Comment by Cruze03** [comment]
Upload whole error log. Some part is cut off. I'll try to fix!

**Comment by reid123455** [comment]
> Upload whole error log. Some part is cut off. I'll try to fix!
> and make sure you're using latest version of plugin.

Sorry for very late reply, because too much problems and busy,
I have been a long time not maintained my CS:GO server

I update my CS:GO server, the crashing problem seems to be fixed
But plugin is still little buggy and has some little problems even I update to last version of this plugin

I record it with timeline, due to my bad English, I guess I can't explain clearly lel
https://youtu.be/ms-5gO3qYTI

In the end, here is my whole error log, hope there is some helpful and useful information

```
L 01/14/2020 - 19:50:48: SourceMod error session started
L 01/14/2020 - 19:50:48: Info (map "de_dust2") (file "C:\Server\csgo server\csgo\addons\sourcemod\logs\errors_20200114.log")
L 01/14/2020 - 19:50:48: [CRASH] Accelerator uploaded crash dump: Crash ID: SVC3-3ACL-L4A2
L 01/14/2020 - 19:50:53: Error log file session closed.
L 01/14/2020 - 19:51:36: SourceMod error session started
L 01/14/2020 - 19:51:36: Info (map "workshop/808512125/de_dust2rework") (file "C:\Server\csgo server\csgo\addons\sourcemod\logs\errors_20200114.log")
L 01/14/2020 - 19:51:36: [SM] Exception reported: invalid handle 0 (error: 4)
L 01/14/2020 - 19:51:36: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:51:36: [SM] Call stack trace:
L 01/14/2020 - 19:51:36: [SM]   [0] ReadFileLine
L 01/14/2020 - 19:51:36: [SM]   [1] Line 108, /home/travis/build/shanapu/MyStore/addons/sourcemod/scripting/include/smartdm.inc::Downloader_GetMaterialsFromVMT
L 01/14/2020 - 19:51:36: [SM]   [2] Line 197, /home/travis/build/shanapu/MyStore/addons/sourcemod/scripting/include/smartdm.inc::Downloader_AddFileToDownloadsTable
L 01/14/2020 - 19:51:36: [SM]   [3] Line 188, /home/travis/build/shanapu/MyStore/addons/sourcemod/scripting/include/smartdm.inc::Downloader_AddFileToDownloadsTable
L 01/14/2020 - 19:51:36: [SM]   [4] Line 139, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_OnMapStart
L 01/14/2020 - 19:51:36: [SM]   [6] Call_Finish
L 01/14/2020 - 19:51:36: [SM]   [7] Line 758, addons/sourcemod/scripting/mystore_core.sp::OnMapStart
L 01/14/2020 - 19:51:55: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:51:55: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:51:55: [SM] Call stack trace:
L 01/14/2020 - 19:51:55: [SM]   [0] CreateTimer
L 01/14/2020 - 19:51:55: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn
L 01/14/2020 - 19:51:55: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:51:55: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:51:55: [SM] Call stack trace:
L 01/14/2020 - 19:51:55: [SM]   [0] CreateTimer
L 01/14/2020 - 19:51:55: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn
L 01/14/2020 - 19:52:09: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:52:09: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:52:09: [SM] Call stack trace:
L 01/14/2020 - 19:52:09: [SM]   [0] CreateTimer
L 01/14/2020 - 19:52:09: [SM]   [1] Line 253, addons/sourcemod/scripting/mystore_item_playermodel.sp::SetClientModel
L 01/14/2020 - 19:52:09: [SM]   [2] Line 181, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_Equip
L 01/14/2020 - 19:52:09: [SM]   [4] Call_Finish
L 01/14/2020 - 19:52:09: [SM]   [5] Line 3509, addons/sourcemod/scripting/mystore_core.sp::UseItem
L 01/14/2020 - 19:52:09: [SM]   [6] Line 1728, addons/sourcemod/scripting/mystore_core.sp::MenuHandler_Preview
L 01/14/2020 - 19:52:57: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:52:57: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:52:57: [SM] Call stack trace:
L 01/14/2020 - 19:52:57: [SM]   [0] CreateTimer
L 01/14/2020 - 19:52:57: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn
L 01/14/2020 - 19:53:23: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:53:23: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:53:23: [SM] Call stack trace:
L 01/14/2020 - 19:53:23: [SM]   [0] CreateTimer
L 01/14/2020 - 19:53:23: [SM]   [1] Line 253, addons/sourcemod/scripting/mystore_item_playermodel.sp::SetClientModel
L 01/14/2020 - 19:53:23: [SM]   [2] Line 181, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_Equip
L 01/14/2020 - 19:53:23: [SM]   [4] Call_Finish
L 01/14/2020 - 19:53:23: [SM]   [5] Line 3509, addons/sourcemod/scripting/mystore_core.sp::UseItem
L 01/14/2020 - 19:53:23: [SM]   [6] Line 1951, addons/sourcemod/scripting/mystore_core.sp::MenuHandler_Item
L 01/14/2020 - 19:53:29: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:53:29: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:53:29: [SM] Call stack trace:
L 01/14/2020 - 19:53:29: [SM]   [0] CreateTimer
L 01/14/2020 - 19:53:29: [SM]   [1] Line 253, addons/sourcemod/scripting/mystore_item_playermodel.sp::SetClientModel
L 01/14/2020 - 19:53:29: [SM]   [2] Line 181, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_Equip
L 01/14/2020 - 19:53:29: [SM]   [4] Call_Finish
L 01/14/2020 - 19:53:29: [SM]   [5] Line 3509, addons/sourcemod/scripting/mystore_core.sp::UseItem
L 01/14/2020 - 19:53:29: [SM]   [6] Line 1951, addons/sourcemod/scripting/mystore_core.sp::MenuHandler_Item
L 01/14/2020 - 19:54:52: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:54:52: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:54:52: [SM] Call stack trace:
L 01/14/2020 - 19:54:52: [SM]   [0] CreateTimer
L 01/14/2020 - 19:54:52: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn
L 01/14/2020 - 19:55:10: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:55:10: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:55:10: [SM] Call stack trace:
L 01/14/2020 - 19:55:10: [SM]   [0] CreateTimer
L 01/14/2020 - 19:55:10: [SM]   [1] Line 253, addons/sourcemod/scripting/mystore_item_playermodel.sp::SetClientModel
L 01/14/2020 - 19:55:10: [SM]   [2] Line 181, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_Equip
L 01/14/2020 - 19:55:10: [SM]   [4] Call_Finish
L 01/14/2020 - 19:55:10: [SM]   [5] Line 3509, addons/sourcemod/scripting/mystore_core.sp::UseItem
L 01/14/2020 - 19:55:10: [SM]   [6] Line 1951, addons/sourcemod/scripting/mystore_core.sp::MenuHandler_Item
L 01/14/2020 - 19:55:17: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:55:17: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:55:17: [SM] Call stack trace:
L 01/14/2020 - 19:55:17: [SM]   [0] CreateTimer
L 01/14/2020 - 19:55:17: [SM]   [1] Line 253, addons/sourcemod/scripting/mystore_item_playermodel.sp::SetClientModel
L 01/14/2020 - 19:55:17: [SM]   [2] Line 181, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_Equip
L 01/14/2020 - 19:55:17: [SM]   [4] Call_Finish
L 01/14/2020 - 19:55:17: [SM]   [5] Line 3509, addons/sourcemod/scripting/mystore_core.sp::UseItem
L 01/14/2020 - 19:55:17: [SM]   [6] Line 1951, addons/sourcemod/scripting/mystore_core.sp::MenuHandler_Item
L 01/14/2020 - 19:55:30: Error log file session closed.
L 01/14/2020 - 19:55:50: SourceMod error session started
L 01/14/2020 - 19:55:50: Info (map "de_dust2") (file "C:\Server\csgo server\csgo\addons\sourcemod\logs\errors_20200114.log")
L 01/14/2020 - 19:55:50: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:55:50: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:55:50: [SM] Call stack trace:
L 01/14/2020 - 19:55:50: [SM]   [0] CreateTimer
L 01/14/2020 - 19:55:50: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn
L 01/14/2020 - 19:55:51: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:55:51: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:55:51: [SM] Call stack trace:
L 01/14/2020 - 19:55:51: [SM]   [0] CreateTimer
L 01/14/2020 - 19:55:51: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn
L 01/14/2020 - 19:56:05: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:56:05: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:56:05: [SM] Call stack trace:
L 01/14/2020 - 19:56:05: [SM]   [0] CreateTimer
L 01/14/2020 - 19:56:05: [SM]   [1] Line 253, addons/sourcemod/scripting/mystore_item_playermodel.sp::SetClientModel
L 01/14/2020 - 19:56:05: [SM]   [2] Line 181, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_Equip
L 01/14/2020 - 19:56:05: [SM]   [4] Call_Finish
L 01/14/2020 - 19:56:05: [SM]   [5] Line 3509, addons/sourcemod/scripting/mystore_core.sp::UseItem
L 01/14/2020 - 19:56:05: [SM]   [6] Line 1728, addons/sourcemod/scripting/mystore_core.sp::MenuHandler_Preview
L 01/14/2020 - 19:56:12: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:56:12: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:56:12: [SM] Call stack trace:
L 01/14/2020 - 19:56:12: [SM]   [0] CreateTimer
L 01/14/2020 - 19:56:12: [SM]   [1] Line 253, addons/sourcemod/scripting/mystore_item_playermodel.sp::SetClientModel
L 01/14/2020 - 19:56:12: [SM]   [2] Line 181, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_Equip
L 01/14/2020 - 19:56:12: [SM]   [4] Call_Finish
L 01/14/2020 - 19:56:12: [SM]   [5] Line 3509, addons/sourcemod/scripting/mystore_core.sp::UseItem
L 01/14/2020 - 19:56:12: [SM]   [6] Line 1951, addons/sourcemod/scripting/mystore_core.sp::MenuHandler_Item
L 01/14/2020 - 19:56:22: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:56:22: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:56:22: [SM] Call stack trace:
L 01/14/2020 - 19:56:22: [SM]   [0] CreateTimer
L 01/14/2020 - 19:56:22: [SM]   [1] Line 253, addons/sourcemod/scripting/mystore_item_playermodel.sp::SetClientModel
L 01/14/2020 - 19:56:22: [SM]   [2] Line 181, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_Equip
L 01/14/2020 - 19:56:22: [SM]   [4] Call_Finish
L 01/14/2020 - 19:56:22: [SM]   [5] Line 3509, addons/sourcemod/scripting/mystore_core.sp::UseItem
L 01/14/2020 - 19:56:22: [SM]   [6] Line 1951, addons/sourcemod/scripting/mystore_core.sp::MenuHandler_Item
L 01/14/2020 - 19:56:29: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:56:29: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:56:29: [SM] Call stack trace:
L 01/14/2020 - 19:56:29: [SM]   [0] CreateTimer
L 01/14/2020 - 19:56:29: [SM]   [1] Line 253, addons/sourcemod/scripting/mystore_item_playermodel.sp::SetClientModel
L 01/14/2020 - 19:56:29: [SM]   [2] Line 181, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_Equip
L 01/14/2020 - 19:56:29: [SM]   [4] Call_Finish
L 01/14/2020 - 19:56:29: [SM]   [5] Line 3509, addons/sourcemod/scripting/mystore_core.sp::UseItem
L 01/14/2020 - 19:56:29: [SM]   [6] Line 1951, addons/sourcemod/scripting/mystore_core.sp::MenuHandler_Item
L 01/14/2020 - 19:56:31: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:56:31: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:56:31: [SM] Call stack trace:
L 01/14/2020 - 19:56:31: [SM]   [0] CreateTimer
L 01/14/2020 - 19:56:31: [SM]   [1] Line 253, addons/sourcemod/scripting/mystore_item_playermodel.sp::SetClientModel
L 01/14/2020 - 19:56:31: [SM]   [2] Line 181, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_Equip
L 01/14/2020 - 19:56:31: [SM]   [4] Call_Finish
L 01/14/2020 - 19:56:31: [SM]   [5] Line 3509, addons/sourcemod/scripting/mystore_core.sp::UseItem
L 01/14/2020 - 19:56:31: [SM]   [6] Line 1951, addons/sourcemod/scripting/mystore_core.sp::MenuHandler_Item
L 01/14/2020 - 19:56:47: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:56:47: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:56:47: [SM] Call stack trace:
L 01/14/2020 - 19:56:47: [SM]   [0] CreateTimer
L 01/14/2020 - 19:56:47: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn
L 01/14/2020 - 19:56:53: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:56:53: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:56:53: [SM] Call stack trace:
L 01/14/2020 - 19:56:53: [SM]   [0] CreateTimer
L 01/14/2020 - 19:56:53: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn
L 01/14/2020 - 19:57:00: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:57:00: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:57:00: [SM] Call stack trace:
L 01/14/2020 - 19:57:00: [SM]   [0] CreateTimer
L 01/14/2020 - 19:57:00: [SM]   [1] Line 253, addons/sourcemod/scripting/mystore_item_playermodel.sp::SetClientModel
L 01/14/2020 - 19:57:00: [SM]   [2] Line 181, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_Equip
L 01/14/2020 - 19:57:00: [SM]   [4] Call_Finish
L 01/14/2020 - 19:57:00: [SM]   [5] Line 3509, addons/sourcemod/scripting/mystore_core.sp::UseItem
L 01/14/2020 - 19:57:00: [SM]   [6] Line 1951, addons/sourcemod/scripting/mystore_core.sp::MenuHandler_Item
L 01/14/2020 - 19:57:03: [SM] Exception reported: Could not create timer, no more handles
L 01/14/2020 - 19:57:03: [SM] Blaming: mystore_item_playermodel.smx
L 01/14/2020 - 19:57:03: [SM] Call stack trace:
L 01/14/2020 - 19:57:03: [SM]   [0] CreateTimer
L 01/14/2020 - 19:57:03: [SM]   [1] Line 253, addons/sourcemod/scripting/mystore_item_playermodel.sp::SetClientModel
L 01/14/2020 - 19:57:03: [SM]   [2] Line 181, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_Equip
L 01/14/2020 - 19:57:03: [SM]   [4] Call_Finish
L 01/14/2020 - 19:57:03: [SM]   [5] Line 3509, addons/sourcemod/scripting/mystore_core.sp::UseItem
L 01/14/2020 - 19:57:03: [SM]   [6] Line 1951, addons/sourcemod/scripting/mystore_core.sp::MenuHandler_Item
```

**Comment by dzinks2009** [comment]
I fixed my issue by setting sv_hibernation to 0

**Comment by Ayrton09** [comment]
with the latest version i have this bug, when i set a skin at the next round disappears

and not fixed with sv_hibernate_when_empty to 0

INFO:
CSGO on Windows
latest sourcemod 1.10
and latest store plugin  27/01/20 - 04:11:04 | b9fed6d | MyStore-0.1.37-b9fed6d




L 01/27/2020 - 09:00:52: [SM]   [0] CreateTimer
L 01/27/2020 - 09:00:52: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn
L 01/27/2020 - 09:00:53: [SM] Exception reported: Could not create timer, no more handles
L 01/27/2020 - 09:00:53: [SM] Blaming: mystore_item_playermodel.smx
L 01/27/2020 - 09:00:53: [SM] Call stack trace:
L 01/27/2020 - 09:00:53: [SM]   [0] CreateTimer
L 01/27/2020 - 09:00:53: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn
L 01/27/2020 - 09:00:59: Error log file session closed.
L 01/27/2020 - 09:01:35: SourceMod error session started
L 01/27/2020 - 09:01:35: Info (map "35hp_2_2015") (file "csgo\addons\sourcemod\logs\errors_20200127.log")
L 01/27/2020 - 09:01:35: [SM] Exception reported: Could not create timer, no more handles
L 01/27/2020 - 09:01:35: [SM] Blaming: mystore_item_playermodel.smx
L 01/27/2020 - 09:01:35: [SM] Call stack trace:
L 01/27/2020 - 09:01:35: [SM]   [0] CreateTimer
L 01/27/2020 - 09:01:35: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn
L 01/27/2020 - 09:01:35: [SM] Exception reported: Could not create timer, no more handles
L 01/27/2020 - 09:01:35: [SM] Blaming: mystore_item_playermodel.smx
L 01/27/2020 - 09:01:35: [SM] Call stack trace:
L 01/27/2020 - 09:01:35: [SM]   [0] CreateTimer
L 01/27/2020 - 09:01:35: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn
L 01/27/2020 - 09:01:55: [SM] Exception reported: Could not create timer, no more handles
L 01/27/2020 - 09:01:55: [SM] Blaming: mystore_item_playermodel.smx
L 01/27/2020 - 09:01:55: [SM] Call stack trace:
L 01/27/2020 - 09:01:55: [SM]   [0] CreateTimer
L 01/27/2020 - 09:01:55: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn
L 01/27/2020 - 09:02:00: [SM] Exception reported: Could not create timer, no more handles
L 01/27/2020 - 09:02:00: [SM] Blaming: mystore_item_playermodel.smx
L 01/27/2020 - 09:02:00: [SM] Call stack trace:
L 01/27/2020 - 09:02:00: [SM]   [0] CreateTimer
L 01/27/2020 - 09:02:00: [SM]   [1] Line 253, addons/sourcemod/scripting/mystore_item_playermodel.sp::SetClientModel
L 01/27/2020 - 09:02:00: [SM]   [2] Line 181, addons/sourcemod/scripting/mystore_item_playermodel.sp::PlayerModels_Equip
L 01/27/2020 - 09:02:00: [SM]   [4] Call_Finish
L 01/27/2020 - 09:02:00: [SM]   [5] Line 3506, addons/sourcemod/scripting/mystore_core.sp::UseItem
L 01/27/2020 - 09:02:00: [SM]   [6] Line 1960, addons/sourcemod/scripting/mystore_core.sp::MenuHandler_Item
L 01/27/2020 - 09:02:18: [SM] Exception reported: Could not create timer, no more handles
L 01/27/2020 - 09:02:18: [SM] Blaming: mystore_item_playermodel.smx
L 01/27/2020 - 09:02:18: [SM] Call stack trace:
L 01/27/2020 - 09:02:18: [SM]   [0] CreateTimer
L 01/27/2020 - 09:02:18: [SM]   [1] Line 224, addons/sourcemod/scripting/mystore_item_playermodel.sp::Event_PlayerSpawn

**Comment by shanapu** [comment]
Hi all and thank you for testing MyStore!
I am very pleased about your commitment, and that motivates me greatly!

> I tried various Store plugins, but some of those have not maintained anymore

> Did you find a fix?

> But plugin is still little buggy and has some little problems even I update to last version of this plugin

> I'll try to fix

**First I think it's necessary to mention that MyStore is still in the development phase and far away from a final or stable release!**
Any help, be it in the form of bug reporting or code contribution, is highly appreciated and needed.
But I recommend **do not use** MyStore **on a productive/public server**.

#### About the issue:
Right now I don't find the leaking Handles (`Could not create timer, no more handles`) or the source of this error. Any help is really appreciated and needed here.

Even though I am working on other parts of the project in the meantime, this topic is not forgotten and has high priority for me.

**Comment by Ayrton09** [comment]
Thank you for working in this hard project, ill really appreciate it!, if i found something i will help as i can.

**Comment by Kxnrl** [comment]
try sm_dump_handles command

**Comment by raytzou** [comment]
thank you all guys' hard working
I still test this plugin sometime with newest version
finally, I found what caused server side crash(precache error) or client side crash
I saw some items setting with "plans", so I thought I can set skin item with time duration like PSE ZombieEscape server
but this function seems to cause precaching error or let client side crash if setting duration plan on skin item

the second problem of my server is that some skins will reset to default skin on next round or change to next map
this problem seems to be caused by bad viewmodel
like my video, the Nepgear I recompiled it for removing other server logo before
only this model will be reset, other models are fine

tl;dr
- set plan on skin items will let server or client side be unstable
- if model reset to default one and can't be unequipped, maybe the model has bad viewmodel or other problem
- due to my bad programming skill and knowing less thing about sourcemod, I can't help with timer handle thing, I'm sorry about that :(

In the end, I really like this store plugin anyway, it's the best store plugin I have ever used :)

**Comment by Ayrton09** [comment]
yep is the best store i wish these errores can be fixed :(

**Comment by nuclearsilo583** [comment]
hi i dont know wheither if this issue is solved yet. But i found my way to figure out the possible and effective way to fix this as for me It worked fine after i change some line in the code. Simply I just removed precache and download all the model at map start. Warning remove this mean you need S-Downloader for all the materials, models,.. that need to be precache by itself before using or cause crash due to lack of precaching using models.

This fix also work for error could not create timer, no more handle for some skin cause this bug only appear for a specific models i dont know which one but there somes. With removed precache on every map start will fix this problem

P/S: I dont think this fix is stable yet. Warning. Since i still got many several crash on this module everything work fine except for this one cause crash for some model

Also the current version of playermodel module is not currently supporting the skin's index which has multiple skin in 1 mdl files like original zephyrus store.

public void PlayerModels_OnMapStart()
{
	/*
        for (int i = 0; i < g_iCount; i++)
	{
		PrecacheModel(g_sModel[i], true);
		Downloader_AddFileToDownloadsTable(g_sModel[i]);

		if (g_sArms[i][0] != 0)
		{
			PrecacheModel(g_sArms[i], true);
			Downloader_AddFileToDownloadsTable(g_sArms[i]);
		}
	}
       */
}
